File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/params/get_joint_control_params.cpp
// get_joint_control_params.cpp

#include <iomanip>
#include <iostream>

#include "common/common_def.h"
#include "h10-w/h10w_params.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 获取关节空间控制参数示例 ===" << std::endl;
  H10wParams params;
  JointControlParams get_params;

  if (params.getJointControlParams(get_params) != 0) {
    std::cerr << "获取关节空间控制参数失败" << std::endl;
    return -1;
  }

  std::cout << "获取关节空间控制参数成功，共 " << get_params.size() << " 项"
            << std::endl;
  std::cout << std::fixed << std::setprecision(3);
  for (const auto& item : get_params) {
    const auto& [joint_index, kp, kd, kpv] = item;
    std::cout << "  joint=" << joint_index << ", kp=" << kp << ", kd=" << kd
              << ", kpv=" << kpv << std::endl;
  }

  return 0;
}

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/params/get_joint_control_params.py
#!/usr/bin/env python3

import humanoid_sdk_py
import humanoid_sdk_py.h10w as h10w
import time


def main():
    print("=== 获取关节空间控制参数示例 ===")
    params = h10w.H10wParams()
    time.sleep(1)

    ret, joint_list = params.getJointControlParams()
    if ret != 0 or joint_list is None:
        print("获取关节空间控制参数失败")
        return -1

    print("获取关节空间控制参数成功，共 {} 项".format(len(joint_list)))
    for item in joint_list:
        joint_index, kp, kd, kpv = item
        print(
            "  joint={}, kp={:.3f}, kd={:.3f}, kpv={:.3f}".format(
                joint_index, kp, kd, kpv
            )
        )
    return 0


if __name__ == "__main__":
    exit(main())
